app.ts

```diff
@@ -0,0 +1,366 @@
+	addRows(recordsLink);
+
+	// Display content
+	$(".loader").fadeOut(200);
+	$(".content").fadeIn(200);
+}
+
+
+// Calculate what data should be fetch when page buttons are clicked
+function pageFunction(btnName: string) {
+	let isValid = false
+
+	// Set trim to start of data
+	if (btnName == "first" && state.trimStart != 0) {
+		isValid = true;
+		state.trimStart = 0;
+		state.trimEnd = state.trimStart + state.records - 1;
+	}
+
+	// Set trim to previous data
+	if (btnName == "prev" && state.trimStart != 0) {
+		// If previous page is end of data && there are not enough records to fill window
+		if ((state.trimStart - 1) - (state.records - 1) < 0) {
+			state.trimStart = 0;
+			state.trimEnd = state.trimStart + state.records - 1;
+		}
+		else {
+			state.trimEnd = state.trimStart - 1;
+			state.trimStart = state.trimEnd - state.records + 1;
+		}
+		isValid = true;
+
+	}
+
+	// Set trim to next data
+	if (btnName == "next" && state.trimEnd != RECORDCOUNT - 1) {
+		// If next page is end of data && there are not enough records to fill window
+		if ((RECORDCOUNT - 1) - (state.trimEnd + 1) < state.records) {
+			state.trimEnd = RECORDCOUNT - 1;
+			state.trimStart = state.trimEnd - state.records + 1;
+		}
+		else {
+			state.trimStart = state.trimEnd + 1;
+			state.trimEnd = state.trimStart + state.records - 1;
+		}
+		isValid = true;
+	}
+
+	// Set trim to end of data
+	if (btnName == "last" && state.trimEnd != RECORDCOUNT - 1) {
+		isValid = true;
+		state.trimEnd = RECORDCOUNT - 1;
+		state.trimStart = state.trimEnd - state.records + 1;
+	}
+
+	if (isValid) {
+		loadIntoTable(el("table"));
+	}
+	else {
+		console.log("Grey out");
+	}
+
+}
+
+
+// Code is only triggered once per user input
+const debounce = (fn: Function, ms: number) => {
+	let timeoutId: ReturnType<typeof setTimeout>;
+	return function (this: any, ...args: any[]) {
+		clearTimeout(timeoutId);
+		timeoutId = setTimeout(() => fn.apply(this, args), ms);
+	};
+};
+
+
+// Last resort debounce application for buttons
+function inputFirst() {
+	pageFunction('first');
+}
+function inputPrev() {
+	pageFunction('prev');
+}
+function inputNext() {
+	pageFunction('next');
+}
+function inputLast() {
+	pageFunction('last');
+}
+const clickFirst = debounce(() => inputFirst(), 800);
+const clickPrev = debounce(() => inputPrev(), 800);
+const clickNext = debounce(() => inputNext(), 800);
+const clickLast = debounce(() => inputLast(), 800);
+
+
+// Add/remove rows from table based on resize event of the window
+window.addEventListener("resize", debounce(async () => {
+
+	// Calculate number rows to be added/deleted
+	let newHeight = Math.floor((window.innerHeight - 160) / 40);
+	let diff = newHeight - state.records;
+
+	let start = state.trimEnd + 1;
+	let end = state.trimEnd + diff;
+
+
+	if (diff < 0) {
+		// Delete rows from last page
+		deleteRows(state.countRec, diff);
+
+		state.countRec = state.countRec + diff;
+		state.trimEnd = state.trimEnd + diff;
+		state.records = newHeight;
```

Comment:
`state.records = newHeight;` is a line that gets repeated in all your conditional cases. Try to remove some repetition and optimise your if and else statements to only apply the necessary logic and have a clean look. 

`state.records = newHeight;` can be called after your if/else statements then it is not necessary for the last else.